### ISSUE TITLE
[Logic Apps] GET on table entity that does not exist fails with 409

### ISSUE THREAD

**Issue by laveeshb** [Azure/Azurite#621]
### Which service(blob, file, queue, table) does this issue concern?
Table

### Which version of the Azurite was used?
Latest from `table` branch (e99fa46807a8bc8449be77d3dc1e55a64c32a07c)

### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)
Local npm build, see above.

### What's the Node.js version?
v14.15.0

### What problem was encountered?
`GET` on a table entity which does not exist returns 409 rather than 404. Our code uses Storage SDK v9.3.3 to make these requests.

Here is the HTTP request our code made for an entity in a table against `Azurite`.

Request:
```
GET http://127.0.0.1:10002/devstoreaccount1/flowa49f8c2bfdc52fa30b3a686dd09237flows(PartitionKey='8870F',RowKey='1B3D3E4BBCA8470EB881907D925956B7_FLOWLATEST')?sv=2018-03-28&sig=zvCkwoKB3FL9bdasgCQr%2B3FPh0yI8HjIqkE2ZxQwSlA%3D&se=2022-11-12T20%3A06%3A14Z&srt=sco&ss=bqt&sp=racupwdl&timeout=15 HTTP/1.1
Host: 127.0.0.1:10002
User-Agent: ResourceStack/6.0.0.1188
x-ms-version: 2018-03-28
x-ms-client-request-id: d25130cc-91eb-4d4d-a5b2-65805f489896
Accept: application/json; odata=minimalmetadata
```

Response:
```
HTTP/1.1 409 Conflict
Server: Azurite-Table/3.9.0-table-alpha.1
x-ms-error-code: EntityDoesNotExist
x-ms-request-id: 69795f6e-daee-47b7-b503-67e194684557
x-ms-version: 2019-12-12
content-type: application/xml
Date: Thu, 12 Nov 2020 20:06:16 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked

fd
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Error>
  <Code>EntityDoesNotExist</Code>
  <Message>The entity to update doesn't exist in the table.
RequestId:69795f6e-daee-47b7-b503-67e194684557
Time:2020-11-12T20:06:16.216Z</Message>
</Error>
0
```

When I run the same code against `Azure storage emulator`, such request fails with 404. Here is an example
```
GET http://127.0.0.1:10002/devstoreaccount1/flowa49f8c2bfdc52fa7879349f76b9546flows(PartitionKey='1171E',RowKey='7149356487AA4BE1894C28003B9B9E84_FLOWLATEST')?sv=2018-03-28&sig=WqjJ2feoWqoFkOCfxdD668JylXz6YdJ6KmJckPsNbe8%3D&se=2022-11-12T20%3A23%3A03Z&srt=sco&ss=bqt&sp=racupwdl&timeout=15 HTTP/1.1
Host: 127.0.0.1:10002
User-Agent: ResourceStack/6.0.0.1188
x-ms-version: 2018-03-28
x-ms-client-request-id: 0b7f3308-601e-4d11-9018-c6d75ccff914
Accept: application/json; odata=minimalmetadata
```

Response:
```
HTTP/1.1 404 Not Found
Cache-Control: no-cache
Transfer-Encoding: chunked
Content-Type: application/json;odata=minimalmetadata;streaming=true;charset=utf-8
Server: Windows-Azure-Table/1.0 Microsoft-HTTPAPI/2.0
x-ms-request-id: 1d8b875d-65aa-415f-82a7-5584c1afed0d
x-ms-version: 2018-03-28
X-Content-Type-Options: nosniff
Date: Thu, 12 Nov 2020 20:23:08 GMT

CA
{"odata.error":{"code":"ResourceNotFound","message":{"lang":"en-US","value":"The specified resource does not exist.\nRequestId:1d8b875d-65aa-415f-82a7-5584c1afed0d\nTime:2020-11-12T20:23:09.3906596Z"}}}
0
```

### Steps to reproduce the issue?
Please see the sample requests above

### Have you found a mitigation/solution?
No

**Comment by XiaoningLiu** [comment]
Hi @laveeshb, thanks for bringing this. Fix included in #622

**Comment by laveeshb** [comment]
@XiaoningLiu thanks for the fix. It works now.
